<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Committee Meeting #2</title>
    <meta charset="utf-8" />
    <meta name="author" content="Nick Strayer" />
    <meta name="date" content="2020-03-02" />
    <link href="libs/remark-css/default.css" rel="stylesheet" />
    <link rel="stylesheet" href="styles.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Committee Meeting #2
### Nick Strayer
### Vanderbilt Biostatistics Department
### 2020-03-02

---





# Slides Layout

![:space 4]

.side-bullet.big-left.middle[

Final paper plan

- First paper
- Second paper
- Third Paper

]


![:space 4]

.side-bullet.big-left.middle[

After graduation plans


]

![:space 4]


.side-bullet.big-left.middle[

Timeline


- The goal
- Potential challenges
- Why this timeline
 
]

---

# Paper \#1: Phewas ME (Submitted to Bioinformatics)

- Photo of paper in "final format"
- Photo of submission confirmation
- Photo of email from researcher using 
- Photo of presentation at RStudio conference

---

# Paper \#2: Consensus SBM: An uncertainty aware network clustering method



.side-bullet.big-left.topics[
SBMs

- Cluster network data
- Flexible in format ({bi, tri, ...}-partite networks)
- Interpretable
]


.side-bullet.big-left.topics.nopad[
### +
]

.side-bullet.big-left.topics[
### Consensus Clustering

- Recognizes that many clusterings may fit data
- Reduce data to pairwise "probability of being in same cluster" distances
]



.side-bullet.big-left.topics.nopad[
### =
]



.side-bullet.big-left.topics[
### C-SBM
- Use posterior to quickly sample lots of potential partitionings
- Consensus clustering provides window to view super high-dimensional results
]

---
class: middle, center

## Performance on simulated data

### [`sbmR` Vignette](http://nickstrayer.me/sbmR/articles/sampling_from_posterior.html)

---

## What does this contribute?


![:space 8]

.side-bullet[
__Statistically principled method of clustering data that defies common probability functions__


_What is the distribution of an entire genome or phenome?_
]

![:space 4]


.side-bullet[
__A new way of looking at the results of bayesian SBMs__

_Previous results either ignored or used summary stats_
]

![:space 4]
  
.side-bullet[
__An easy jumping off point for specialization into more specialized methods__

_Polypartite specialization, Time-dependent, ..._
]


---
class: center, middle

# How can it be expanded?

---

### Polypartite specialization

.side-bullet[
Place restrictions on potential connections

Can greatly increase the models power to resolve structure
]

.side-bullet[
`sbmR` package allows defining complex relationships between nodes

I am not aware of any other software doing this
]

![:space 4]

.center-big[E.g. __Genotype__ - __Patient__ - __Phenotype__]

![tripartite_example]()
  
---


### Multiple edge types

- What if the "network" has different edge types? 
- E.g. ICD9 counts and protein fold changes
- SBM fitting protocols could be modified to deal with this
  

![different_edge_types]()


---
  
### Time dependence
  
- How do comorbidities or protein pathways change over time?
- Clusters remain the same, edge propensities change. 
- See [Role Detection in Bicycle-Sharing Networks Using Multilayer Stochastic Block Models](https://arxiv.org/abs/1908.09440) by Carlen Et Al

  
![time_dependent_sbm]()
  

---
class: middle 

## Potential journals

- Nature Methods
- American Statistician
- Bioinformatics
- ...?


---


# Paper \# 3: sbmR: An R package for fast and robust fitting and investigating SBM models in R.

![sbmr_site]()

---

## Effort report (by the numbers)

.side-bullet.big-left.middle[
1,100+

[Git commits](https://github.com/nstrayer/sbmR/commits/master)
]

.side-bullet.big-left.middle[
842

Unit tests ([171 R tests](https://github.com/nstrayer/sbmR/tree/master/tests/testthat), [671 C++ tests](https://github.com/nstrayer/sbmR/tree/master/src/cpp_tests))

]


.side-bullet.big-left.middle[
33

[Fully documented functions](http://nickstrayer.me/sbmR/reference/index.html)
]

.side-bullet.big-left.middle[
2

Invited conference presentations
]


.side-bullet.big-left.middle[
### 1

[Package website](http://nickstrayer.me/sbmR)
]




---

## Features beyond consensus SBMs


.side-bullet.big-left.middle[
[New more efficient agglomerative merging technique](http://nickstrayer.me/sbmR/articles/agglomerative_merging.html)

- Cuts runtime of original algorithm down by roughly a factor of `N`
- Demos of superior performance in package vignettes
]


.side-bullet.big-left.middle[
[Advanced node and edge type specification](http://nickstrayer.me/sbmR/reference/new_sbm_network.html#examples)

- No other package that I am aware of does this
- Enables semi-supervised approaches etc. 
]
  


---
class: middle

## Potential journals

- R Journal
- Bioinfomatics (application note)

---
class: four-square

# After graduation plans


### Hopkins Postdoc
### Hopkins Data Scientist
### LA Times Data Scientist
### Other places to consider?


---
class: middle

# Timeline


![](figs/second_meeting_timeline.png)


---
# Potential challenges

&gt; Yaomin quote about ambition


- Slow reviews
- Long edits
- Collaboration work


---
# Why this timeline

- My wife is targeting the same time
- Job opening/acceptance times
- I've been here a while...
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script src="macros.js"></script>
<script>var slideshow = remark.create({
"ratio": "16:9",
"highlightStyle": "github",
"highlightLines": true,
"slideNumberFormat": "#%current% of %total%"
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
